File: java/src/demos/dlineage/dataflow/model/xml/PositionUtil.java

package demos.dlineage.dataflow.model.xml;

import demos.dlineage.util.Pair;

public class PositionUtil
{

	public static Pair<Integer, Integer> getStartPos( String coordinate )
	{
		if ( coordinate != null )
		{
			String[] splits = coordinate.replace( "[", "" )
					.replace( "]", "" )
					.split( "," );
			if ( splits.length == 4 )
			{
				return new Pair<Integer, Integer>( Integer.parseInt( splits[0].trim( ) ),
						Integer.parseInt( splits[1].trim( ) ) );
			}
		}
		return null;
	}

	public static Pair<Integer, Integer> getEndPos( String coordinate )
	{
		if ( coordinate != null )
		{
			String[] splits = coordinate.replace( "[", "" )
					.replace( "]", "" )
					.split( "," );
			if ( splits.length == 4 )
			{
				return new Pair<Integer, Integer>( Integer.parseInt( splits[2].trim( ) ),
						Integer.parseInt( splits[3].trim( ) ) );
			}
		}
		return null;
	}
}

File: dotnet/src/demos/dataFlowAnalyzer/dataflow/model/xml/PositionUtil.cs
namespace gudusoft.gsqlparser.demos.dlineage.dataflow.model.xml
{
    using System;

    public class PositionUtil
	{

        public static Tuple<int, int> getStartPos(string coordinate)
        {
            if (!string.ReferenceEquals(coordinate, null))
            {
                string[] splits = coordinate.Replace("(", "").Replace(")", "").Split(',');
                if (splits.Length == 4)
                {
                    return new Tuple<int, int>(int.Parse(splits[0].Trim()), int.Parse(splits[1].Trim()));
                }
            }
            return null;
        }

        public static Tuple<int, int> getEndPos(string coordinate)
        {
            if (!string.ReferenceEquals(coordinate, null))
            {
                string[] splits = coordinate.Replace("(", "").Replace(")", "").Split(',');
                if (splits.Length == 4)
                {
                    return new Tuple<int, int>(int.Parse(splits[2].Trim()), int.Parse(splits[3].Trim()));
                }
            }
            return null;
        }
    }

}